src/Checkbox.js

```diff
@@ -0,0 +1,118 @@
+import React from 'react'
+import styled, { css } from 'styled-components'
+import PropTypes from 'prop-types'
+import { Icon, Flex, Box, Label, theme } from '..'
+
+class Checkbox extends React.Component {
+  handleClick(e) {
+    const { name, value, onClick } = this.props
+    const checked = e.target.checked
+    onClick(name, value, checked)
+  }
+
+  render() {
+    const {
+      name,
+      id,
+      disabled,
+      size,
+      fontSize,
+      children,
+      onClick,
+      checked,
+      defaultChecked,
+      theme: themeProp,
+      ...restProps
+    } = this.props
+    const currentTheme = themeProp || theme
+    return (
+      <StyledLabel
+        htmlFor={name}
+        data-test="design-system-checkbox"
+        disabled={disabled}
+        theme={currentTheme}
+      >
+        <input
+          type="checkbox"
+          name={name}
+          id={this.props.id || name}
+          disabled={disabled}
+          defaultChecked={checked || defaultChecked}
+          {...restProps}
+          onChange={e => this.handleClick(e)}
+          style={{ opacity: 0, zIndex: -1 }}
+        />
+        <StyledCheckbox
+          data-test="checkbox-facade"
+          disabled={disabled}
+          size={size}
+          theme={currentTheme}
```

Comment:
Same here, I don't see the need to import `theme` and pass it to your styled component. `ThemeProvider` should take of that for you. If you would like to import `theme` outside of a styled component you can wrap your component with a `withTheme` HOC.